test/integration/apiserver/flowcontrol/fight_test.go

```diff
@@ -0,0 +1,316 @@
+			ft.t.Logf("For invert=%v, i=%v, notified of %#+v", invert, i, obj)
+			obj = peel(obj)
+			switch typed := obj.(type) {
+			case *flowcontrol.FlowSchema:
+				key, _ := cache.MetaNamespaceKeyFunc(obj)
+				ft.rvMutex.Lock()
+				defer ft.rvMutex.Unlock()
+				ft.notifiedRVs[invert][i][key] = typed.ResourceVersion
+			}
+		},
+		AsFieldManager:         fieldMgr,
+		FoundToDangling:        foundToDangling,
+		InformerFactory:        informerFactory,
+		FlowcontrolClient:      fcIfc,
+		ServerConcurrencyLimit: 200,         // server concurrency limit
+		RequestWaitLimit:       time.Minute, // request wait limit
+		ObsPairGenerator:       metrics.PriorityLevelConcurrencyObserverPairGenerator,
+		QueueSetFactory:        fqtesting.NewNoRestraintFactory(),
+	})
+	ft.ctlrs[invert][i] = ctlr
+	informerFactory.Start(ft.stopCh)
+	if ctlr.WaitForCacheSync(ft.stopCh) {
+		ft.t.Logf("Achieved initial sync for invert=%v, i=%v", invert, i)
+	} else {
+		ft.t.Fatalf("Never achieved initial sync for invert=%v, i=%v", invert, i)
+	}
+}
+
+func (ft *fightTest) waitForLastRVs() {
+	AOK := false
+	// wait until notifiedRVs[invert][i] covers lastRVs for all invert, i
+	for k := 1; k < 11 && !AOK; k++ {
+		ft.t.Logf("For size=%d, iteration=%d, k=%d, starting to wait for lastRVs=%v", ft.size, ft.iteration, k, ft.lastRVs)
+		time.Sleep(time.Millisecond * time.Duration(k*100))
+		AOK = true
+		ft.foreach(func(invert bool, i int) {
+			ft.rvMutex.Lock()
+			defer ft.rvMutex.Unlock()
+			for key, rv := range ft.lastRVs {
+				if ft.notifiedRVs[invert][i][key] != rv {
+					AOK = false
+				}
+			}
+		})
+	}
+	if !AOK {
+		func() {
+			ft.rvMutex.Lock()
+			defer ft.rvMutex.Unlock()
+			ft.t.Logf("For size=%d, iteration=%d, lastRVs=%v but notifiedRVs=%#+v", ft.size, ft.iteration, ft.lastRVs, ft.notifiedRVs)
+		}()
+		ft.someTimedOut = true
+	}
+}
+
+func (ft *fightTest) updateAndCheckHistories(invert bool, i int, ctlrRVs map[string]string) int {
+	timeLimit := ft.now.Add(-time.Minute)
+	var nWrites int
+	for key, rv := range ctlrRVs {
+		ft.t.Logf("For invert=%v, i=%v, iteration=%d, wrote to %q", invert, i, ft.iteration, key)
+		ft.lastRVs[key] = rv
+		nWrites++
+		hist := ft.writeHistories[invert][i][key]
+		hist = append(hist, ft.now)
+		for idx, updateTime := range hist {
+			if updateTime.After(timeLimit) {
+				hist = hist[idx:]
+				break
+			}
+		}
+		ft.writeHistories[invert][i][key] = hist
+		if len(hist) > 6 {
+			ft.t.Errorf("For invert=%v, i=%v, iteration=%v, FlowSchema %q, found %d updates in the last minute (%#+v)", invert, i, ft.iteration, key, len(hist), hist)
+		}
+	}
+	return nWrites
+}
+
+func TestConfigConsumerFight(t *testing.T) {
+	// Disable the APF FeatureGate so that the normal config consumer
+	// controller does not interfere
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.APIPriorityAndFairness, false)()
+	_, loopbackConfig, closeFn := setup(t, 100, 100)
+	defer closeFn()
+	const size = 3
```

Comment:
So actually, to clarify. The size is actually the "number of kube-apiservers", right? [or rather I should say, 1/2th of kube-apiserver, because there is second-half with invert, right?].

If so, it might be useful to make name it more explicitly (both here and in the fightTest).

Reply:
Updated comment and name.